### PR TITLE
Feat: move to vue 3 and vite

### LINKED ISSUE #32: Prebuild vue components
For now the import strategy is to just import specific vue-file.

Would make more sense to provide a compiled umd package so that components could be used from different environments and we don't enforce specific bundler + config.

As we go we can also:
- [ ] Check tree shaking works. One solution is creating separate file for every spinner. Another way is to do a [config](https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free). Each of these should be tested with analyzer to make sure bundle is built correctly.

See #13

### LINKED ISSUE #44: FulfillingSquareSpinner doesn't support the animation duration
image

### LINKED ISSUE #45: `beforeDestroy` has been renamed to `beforeUnmount`.
I tried to use it in VUE3, do you have any plans

image

There are some compatibility issues

### LINKED ISSUE #46: Vue.js 3
Anyway to use this with Vue.js 3?

### PR DESCRIPTION
**Pull request by RVitaly1978**
close #32 
close #46 
close #44 
close #45

### CODE REVIEW

**Comment by asvae** [comment]
Could you mention with `close` all relevant issues?

File: .npmignore [hunk 1]
```diff
@@ -1,30 +0,0 @@
-build/
```
Comment (asvae) [inline .npmignore:1]: We should keep npmignore with whatever is irrelevant for build.

File: README.md [hunk 2]
```diff
@@ -17,191 +19,110 @@ We've collected and crafted a rich collection of spinners animated with css whic
...
-          :animation-duration="1000"
-          :size="60"
-          :color="'#ff1d5e'"
-     />
+    <atom-spinner :animation-duration="1000" :size="60" color="#ff1d5e" />
   </div>
 </template>
 
 <script>
-  // To use minified css and js files instead of .vue single file components:
```
Comment (asvae) [inline README.md:42]: Are you sure it's getting tree shaked without file import?

**Review by asvae — APPROVED** [review]
Looks good 👍

**Review by aluarius — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -1,85 +1,130 @@
...
-    "dev:lib": "webpack --config build/webpack.lib.conf.js --watch --progress --hide-modules",
-    "dev:docs": "node build/dev-server.js",
-    "build": "npm run build:lib && npm run build:docs",
-    "build:lib": "node build/build-lib.js",
-    "build:docs": "node build/build.js",
-    "lint": "eslint --ext .js,.vue src"
+    "prepare": "husky install",
+    "dev": "vite",
+    "build": "tsx build/build.ts",
+    "preview": "vite preview",
```
Comment (m0ksem) [inline package.json]: Preview what?

Also, we don't need a favicon.

File: package.json [hunk 4]
```diff
@@ -1,85 +1,130 @@
...
+      "stylelint --fix"
+    ],
+    "*.json": [
+      "prettier --write",
+      "eslint --fix"
+    ]
   },
   "peerDependencies": {
-    "vue": "^2.5.2"
+    "vue": "^3.2.0"
```
Comment (m0ksem) [inline package.json:93]: Must also be in dev deps. For me, vue haven't installed for some reasone in `yarn v1.22.19`.

**Review by m0ksem — APPROVED** [review]

**Comment by m0ksem** [comment]
Also, style.css is not minimized.

File: README.md (hunk 2 above)
Comment (RVitaly1978) [reply README.md:42]: After setting `"sideEffects": ["*.css"]` in `package.json` tree-shaking works in production mode without file import

File: .npmignore (hunk 1 above)
Comment (RVitaly1978) [reply .npmignore:1]: Instead I set `"files": ["dist"]` to `package.json`.

**Comment by bhanu2217** [comment]
When are you planning to have release for Vue.JS 3?

**Review by m0ksem — CHANGES_REQUESTED** [review]
image

styles.css is missing.

**Comment by m0ksem** [comment]
> When are you planning to have release for Vue.JS 3?

After this PR is merged.

**Review by m0ksem — APPROVED** [review]

**Pull request merged**